src/test_utils.rs

```diff
@@ -16,6 +25,8 @@ use crate::key::{self, DcKey};
 pub(crate) struct TestContext {
     pub ctx: Context,
     pub dir: TempDir,
+    /// Counter for [recv_msg] IMAP UID values.
```

Comment:
It is unclear if it is incremented before assigning the UID or not.

Maybe something like "Number of messages received with [recv_msg]. Used to assign fake IMAP UID values."

Reply:
i kind of purposefully didn't document how this actually works.  It's not pub because it's an implementation detail, I made it clearer that it's private to the function.